src/TopBar/index.tsx

```diff
@@ -15,16 +15,28 @@ export interface Props {
   title?: React.ReactNode;
   customLogo?: React.ReactNode;
   onLogoClick?: Function;
+  theme?: TopBarTheme;
+}
+
+export enum TopBarTheme {
+  DEFAULT = "default",
+  PLAIN = "plain",
 }
 
 /**
  * Global page-level header component.
  */
 export class TopBar extends React.PureComponent<Props> {
+  static defaultProps: Partial<Props> = {
```

Comment:
I think we may want to change this type to only include props we have defaults for, eg `static defaultProps: Pick<Props, "theme">`. An issue I was seeing when working on `ResourceTile` is that TypeScript will actually use the type of `defaultProps` to determine which props can be omitted, even if they're normally required. I believe I was seeing that when the type was `Partial<Props>` it considered all props to optional since `defaultProps` would have a key for every prop.

Reply:
ohh, #til - thanks for calling out!